libparsec/crates/platform_http_proxy/src/lib.rs

```diff
@@ -63,50 +56,56 @@ mod native {
 
     impl ProxyConfig {
         /// Set the http proxy to use, will overwrite the previous configuration.
-        pub(crate) fn with_http_proxy<S: AsRef<str>>(mut self, proxy: S) -> Self {
-            self.http_proxy.replace(proxy.as_ref().to_string());
-            self
+        pub(crate) fn with_http_proxy(mut self, proxy: String) -> anyhow::Result<Self> {
```

Comment:
Why requiring an allocated String ? `impl AsRef<str>` or `impl IntoProxyScheme` would be better IMO